packages/react/src/analytics/integrations/shared/GoogleConsentMode/GoogleConsentMode.ts

```diff
@@ -0,0 +1,154 @@
+import { type ConsentData } from '@farfetch/blackout-analytics';
+import {
+  type GoogleConsentCategoryConfig,
+  type GoogleConsentModeConfig,
+  GoogleConsentType,
+} from './types.js';
+import { isEqual, omit } from 'lodash-es';
+
+/**
+ * GoogleConsentMode handles with Google Consent Mode v2.
+ */
+export class GoogleConsentMode {
+  private dataLayer!: string; // Stores different data layer names
+  private config?: GoogleConsentModeConfig; // Stores default or customized consent category mappings
+  private consentConfig!: Record<string, GoogleConsentCategoryConfig>; // Stores default or customized consent category mappings
+  private lastConsent?: Record<
+    string,
+    Array<string> | string | number | undefined
+  >;
+
+  constructor(
+    dataLayer: string,
+    initConsent: ConsentData | null,
+    config?: GoogleConsentModeConfig,
+  ) {
+    this.dataLayer = dataLayer;
+    this.config = config;
+
+    // select only the Google Consent Elements
+    this.consentConfig = omit(this.config || {}, [
```

Comment:
I think this name is misleading since there is a `this.config` variable that is of type GoogleConsentModeConfig. What about renaming this to configExcludingRegionsAndWaitForUpdate? It is a long name but more distinguishable.

Reply:
I think this two options excluded(omitted), was the only one properties not "consent" values.
Could change for example:

export type GoogleConsentModeConfig =
   {
   consent: GoogleConsentMappingsBase<GoogleConsentCategoryConfig>
    regions?: Array<GoogleConsentRegionConfig>;
    wait_for_update?: number;
  };